### DISCUSSION TITLE
Filters are based on polling even when connected via Websocket, right?

### DISCUSSION THREAD

**Discussion by fbit** [web3j/web3j#1432] [Q&A] [answered]
I really like the functional API for filters.

However it seems like event filtering is accomplished via polling in the background. It's not instantly push based, even when I connect to a node via Websocket.

For example this one fires every ~5 seconds instead of popping up instantly when a new block is added:

web3j.blockFlowable(false).subscribe(block -> {
			log.info("NEW BLOCK -> " + block.getBlock().getNumber().intValue());
		});

Or did I miss something?

**Comment by mohamedelshami** [comment, marked as answer]
Yes, is this still polling, subscribing to new blocks is supported in WebSocketService; here is an [example](https://github.com/web3j/web3j/blob/49fe2c4e2d9d325ec465879736d6c384f41a4115/core/src/test/java/org/web3j/protocol/websocket/WebSocketServiceTest.java#L385)

**Reply by fbit** [reply]
Thank you! Is there also a way to listen to contract events via push?
